test/index.spec.ts

```diff
@@ -63,19 +63,31 @@ describe('resin-semver', () => {
 		});
 
 		it('should correctly compare ".dev" versions', () => {
-			expect(semver.compare('2.0.0', '2.0.0+dev')).to.equal(1);
+			expect(semver.compare('2.0.0', '2.0.0+dev')).to.equal(-1);
 			expect(semver.compare('2.0.0', '2.0.0-dev')).to.equal(1);
-			expect(semver.compare('2.0.0', '2.0.0.dev')).to.equal(1);
-			expect(semver.compare('2.0.0+rev6', '2.0.0+rev6.dev')).to.equal(1);
-			expect(semver.compare('2.0.0+rev6.dev', '2.0.0+rev6')).to.equal(-1);
+			expect(semver.compare('2.0.0', '2.0.0.dev')).to.equal(1);// B is invalid
+			expect(semver.compare('2.0.0+rev6', '2.0.0+rev6.dev')).to.equal(-1);
+			expect(semver.compare('2.0.0+rev6.dev', '2.0.0+rev6')).to.equal(1);
 			expect(semver.compare('2.0.0+rev6.dev', '2.0.0+rev6.dev')).to.equal(0);
-			expect(semver.compare('Resin OS 2.0.0+rev3', 'Resin OS 2.0.0+rev3.dev')).to.equal(1);
-			expect(semver.compare('Resin OS 2.0.0+rev3.dev', 'Resin OS 2.0.0+rev3')).to.equal(-1);
+			expect(semver.compare('Resin OS 2.0.0+rev3', 'Resin OS 2.0.0+rev3.dev')).to.equal(-1);
+			expect(semver.compare('Resin OS 2.0.0+rev3.dev', 'Resin OS 2.0.0+rev3')).to.equal(1);
 			expect(semver.compare('Resin OS 2.0.0+rev3.dev', 'Resin OS 2.0.0+rev3.dev')).to.equal(0);
-			expect(semver.compare('2.0.0', 'Resin OS 2.0.0.dev')).to.equal(1);
-			expect(semver.compare('Resin OS 2.0.0.dev', '2.0.0')).to.equal(-1);
+			expect(semver.compare('2.0.0', 'Resin OS 2.0.0.dev')).to.equal(1);// B is invalid
+			expect(semver.compare('Resin OS 2.0.0.dev', '2.0.0')).to.equal(-1);// A is invalid
 			expect(semver.compare('Resin OS 2.0.0.dev', '2.0.0.dev')).to.equal(0);
 		});
+
+		it('should correctly compare "(dev)" and "(prod)" versions without revisions', () => {
+			expect(semver.compare('Resin OS 2.0.0', 'Resin OS 2.0.0 (dev)')).to.equal(-1);
+			expect(semver.compare('Resin OS 2.0.0', 'Resin OS 2.0.0 (prod)')).to.equal(-1);
```

Comment:
I think these two should be equivalent - no variant should implicitly means `prod`, I think?

Reply:
Do we have an answer to this (@imrehg)? I've been assuming in these other comments we'd like the downloads (i.e. descending from best to worst choice) to be sorted as:

    'Resin OS 2.0.0+rev3 (prod)',
    'Resin OS 2.0.0+rev3',
    'Resin OS 2.0.0+rev3 (dev)'

(i.e. dev < anything else).

I'm actually happy to ignore my original comment here: I don't think empty is necessarily equal to `prod`, (i.e. `'' < prod` is fine), but I do think empty should be greater than `dev`.

Reply:
@pimterry @thgreasi 
Ah, so here's the conversation, sorry for leaving comments along the path in this PR.
I think some of the misunderstanding for me comes from the fact that I was considering semver on its own (what should be ranked higher in versions in general), while actually this module supposed to directly determine the order of versions available to download in the dashboard, a much more specific problem.

At first glance, for the general case the  "more specific is higher ranked" should make sense and would argue for it (ie. .dev higher ranked than those that don't have .anything).

In the specific download case, that's clearly not the case, and both .prod and especially .dev are special cases. In this setup, .prod is more specific, .dev should be ranked lower, and no extension is probably considered "like a .prod, but not overtaking an explicit .prod version)"..... Not sure if this becomes more confusing or difficult...

Thus I would say that @pimterry's rundown should be correct in our case of used for resin release ordering purposes, while it represents a policy, thus probably cannot call this a "general" semver ordering. :/

What do you think?

Reply:
Discussed further in flowdock: https://www.flowdock.com/app/rulemotion/resin-frontend/messages/2075644

Conclusion, this (descending) ordering is correct:

```
2.0.0+zzz
2.0.0+prod
2.0.0+abc
2.0.0
2.0.0+something.dev
2.0.0+dev
```

i.e. for metadata, `dev` < none < anything, and we then sort alphabetically within that.